relationships-api/src/main/java/org/commonjava/maven/atlas/graph/jackson/ProjectRelationshipKeyDeserializer.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (C) 2012 Red Hat, Inc. (jdcasey@commonjava.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.maven.atlas.graph.jackson;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.KeyDeserializer;
+
+import java.io.IOException;
+
+/**
+ * Created by ruhan on 3/16/17.
+ */
+@SuppressWarnings( { "rawtypes", "unchecked" } )
+public final class ProjectRelationshipKeyDeserializer
```

Comment:
So this is used for the mapping between PluginRelationship and DependencyRelationships, I guess? Does this result in JSON embedded in JSON...so the key is an embedded JSON string? I wonder if there's a better way to do the mapping, maybe with a composite key of GAV and `managed` flag? You can uniquely identify a Maven plugin declaration using those four fields, and they would be easy to serialize into a string:

```
    <groupId>:<artifactId>:<version>:<managed>
```